addons/survey/survey.py

```diff
@@ -164,11 +164,16 @@ def _get_result_url(self, cr, uid, ids, name, arg, context=None):
             res[survey.id] = urljoin(base_url, "survey/results/%s" % slug(survey))
         return res
 
+    def _is_answered(self, cr, uid, ids, name, arg, context=None):
+        res = {}
+        for user_input in self.pool['survey.user_input'].search_read(cr, uid, [('survey_id', 'in', ids)], ['state'], context=context):
```

Comment:
Could'nt you do a search (survey_id, in ids), (state, =, new) and set those to False, the other one being True ?